internal/blocksync/pool.go

```diff
@@ -130,27 +130,23 @@ func (*BlockPool) OnStop() {}
 
 // spawns requesters as needed
 func (pool *BlockPool) makeRequestersRoutine(ctx context.Context) {
-	for {
-		if !pool.IsRunning() {
-			break
+	for pool.IsRunning() {
+		if ctx.Err() != nil {
```

Comment:
I'd be surprised if these weren't always the same (e.g. if the ctx.Err was ever nil when isRunning was true), based on how the service machinery works.

I'm not sure that matters...

Reply:
It seems like the `BaseService.quit` channel, which determines the result of `IsRunning`, is a separate context: https://github.com/tendermint/tendermint/blob/cf2a00b398cded2235247d3cd1378ec433216233/libs/service/service.go#L118-L123

Reply:
Ah, that's just to make sure that the thread that gets started gets cleaned up reliably: the main context getting canclled calls Stop() (which calls OnStop() and als calls the `bs.cancel` function, and I'm pretty sure these behaviors are correct.